### PR TITLE
SDW Mandatory fields are marked as mandatory

### PR DESCRIPTION
**Pull request by saurabhkg891**
SDW Mandatory fields are marked as mandatory

Mandatory fields are marked as mandatory in SDW module within RAD Lab UI

Issues Fixes: [https://github.com/GoogleCloudPlatform/rad-lab/issues/137](https://github.com/GoogleCloudPlatform/rad-lab/issues/137)

### Asana Task
[https://app.asana.com/0/1202464617347318/1204085111104082/f](https://app.asana.com/0/1202464617347318/1204085111104082/f)

### CODE REVIEW

**Review by guptamukul-google — APPROVED** [review]
Looks good to me.

File: radlab-ui/webapp/src/utils/terraform.ts [hunk 1]
```diff
@@ -113,7 +113,11 @@ const mapHclToUIVar = (
       ? false
       : null,
     // In TF, desription = "" is how we say it's optional
-    required: hclVar.default ? hclVar.default !== "" : false,
+    required: hclVar.default
+      ? hclVar.default !== ""
+      : hclVar.default === undefined
+      ? true
+      : false,
```
Comment (acolver) [inline radlab-ui/webapp/src/utils/terraform.ts]: Can we simplify to `required: typeof hclVar.default === "undefined" || hclVar.default === ""`

File: radlab-ui/webapp/src/components/forms/StepCreator.tsx [hunk 2]
```diff
@@ -46,9 +46,7 @@ const StepCreator: React.FC<StepCreator> = ({ variableList, idx }) => {
     }
 
     // Required checks
-    if (!variable.required) return null
-    if (value || value === "0") return null
-    if (!variable.default) return null
+    if (!variable.required || value || value === "0") return null
```
Comment (acolver) [inline radlab-ui/webapp/src/components/forms/StepCreator.tsx:49]: Do we need 3 checks here if we have new logic (in the `terraform.ts`) that properly set this? Can we just do `if (!variable.required) return null`?

**Review by acolver — CHANGES_REQUESTED** [review]

File: radlab-ui/webapp/src/components/forms/StepCreator.tsx (hunk 2 above)
Comment (saurabhkg891) [reply radlab-ui/webapp/src/components/forms/StepCreator.tsx:49]: it's required we if we keep only `(!variable.required)`  then while typing error message not disappearing

**Review by acolver — APPROVED** [review]

**Pull request merged**